File: python/examples/example2.py
#!/usr/bin/env python
import sys
sys.path.insert(0,'..')
import drqueue
import os

print "Master: %s\n"%(os.environ["DRQUEUE_MASTER"])

computer_list = drqueue.get_computer_list (drqueue.CLIENT)
print "Computers connected to the master:"
for computer in computer_list:
    print "ID: %3i  Name: %s | Enabled: %s"%(computer.hwinfo.id,computer.hwinfo.name.ljust(20),(lambda x: x and "Yes" or "No")(computer.limits.enabled)) 

print "\nJobs registered in the master:"
job_list = drqueue.get_job_list (drqueue.CLIENT)
for job in job_list:
    print "ID: %3i  Name: %s"%(job.id,job.name)
    print "Frames:"
    frame_list = drqueue.get_job_frame_list (job,drqueue.CLIENT)
    for frame_number, frame in frame_list.items() :
        try:
            computer_name = computer_list[frame.icomp].hwinfo.name
        except IndexError:
            computer_name = "None"
        print "\tFrame: %4i  Status: %3i  Computer: %s "%(frame_number, frame.status, computer_name.ljust(20))

File: ruby/example2.rb
#!/usr/bin/ruby


require 'drqueue'


puts "Master: #{ENV["DRQUEUE_MASTER"]}\n"

computer_list = Drqueue::request_computer_list(Drqueue::CLIENT)
puts "Computers connected to the master:"
computer_list.each do |c|
	print "ID: #{c.hwinfo.id} Name: #{c.hwinfo.name} | Enabled: "
	if "#{c.limits.enabled}" == "1"
		print "Yes\n"
	else
		print "No\n"
	end
end

puts "\nJobs registered in the master:"
job_list = Drqueue::request_job_list(Drqueue::CLIENT)
job_list.each do |j|
	puts "ID: #{j.id} Name: #{j.name}"
	print "Frames: "
	frame_list = j.request_frame_list(Drqueue::CLIENT)
	index = 0
	frame_list.each do |f|
		begin
			computer_name = computer_list[f.icomp].hwinfo.name
		rescue
			computer_name = "None"
		end
		puts "\tFrame: #{j.frame_index_to_number(index)} Status: #{f.status} Computer: #{computer_name}"
		index += 1
	end
end